packages/http/src/mocker/HttpMocker.ts

```diff
@@ -49,22 +51,52 @@ export class HttpMocker implements IMocker<IHttpOperation, IHttpRequest, IHttpCo
       negotiationResult = helpers.negotiateOptionsForValidRequest(resource, mockConfig);
     }
 
-    // preparing response body
-    let body;
-    const example = negotiationResult.example;
-
-    if (example && 'value' in example && example.value !== undefined) {
-      body = typeof example.value === 'string' ? example.value : JSON.stringify(example.value);
-    } else if (negotiationResult.schema) {
-      body = await this._exampleGenerator.generate(negotiationResult.schema, negotiationResult.mediaType);
-    }
+    const [body, mockedHeaders] = await Promise.all([
+      computeBody(negotiationResult, this._exampleGenerator),
+      computeMockedHeaders(negotiationResult.headers, this._exampleGenerator),
+    ]);
 
     return {
       statusCode: parseInt(negotiationResult.code),
       headers: {
+        ...mockedHeaders,
         'Content-type': negotiationResult.mediaType,
       },
       body,
     };
   }
 }
+
+function isINodeExample(nodeExample: INodeExample | INodeExternalExample | undefined): nodeExample is INodeExample {
+  return !!nodeExample && 'value' in nodeExample;
+}
+
+function computeMockedHeaders(headers: IHttpHeaderParam[], ex: IExampleGenerator): Promise<Dictionary<string>> {
+  const headerWithPromiseValues = mapValues(keyBy(headers, h => h.name), async header => {
+    if (header.content && header.content.schema) {
+      return ex.generate(header.content.schema, 'application/json');
```

Comment:
Everything makes sense to me apart from this bit, why are we saying that headers are JSON? Does this do anything or is the value essentially ignored because they can only really have strings in headers anyway?

Reply:
Good question. That parameter is indeed useless and it's basically cheating the example generator, which is requiring this parameter and it MUST be application/json otherwise it throws an exception.

https://github.com/stoplightio/prism/blob/master/packages/http/src/mocker/generator/JSONSchemaExampleGenerator.ts#L27

I think I'll just destroy it as part of the work to use OpenAPI Sampler.

Reply:
@philsturgeon I've dropped the bomb: https://github.com/stoplightio/prism/compare/feat/simple-generator?expand=1